squash_wip_test.go

```diff
@@ -0,0 +1,274 @@
+}
+
+func TestSquashWipCommits_failsOnMainBranch(t *testing.T) {
+	output, configuration := localSetup(t)
+	exitedWithCode := -1
+	exit = func(code int) {
+		exitedWithCode = code
+	}
+
+	squashWip(configuration)
+
+	equals(t, 1, exitedWithCode)
+	assertOutputContains(t, output, "Make sure you are on the wip-branch before running quash-wip")
+}
+
+func TestCommitsOnCurrentBranch(t *testing.T) {
+	_, configuration := localSetup(t)
+	createFileAndCommitIt(t, "file1.txt", "irrelevant", "not on branch")
+	start(configuration)
+	createFileAndCommitIt(t, "file2.txt", "irrelevant", "on branch")
+	createFile(t, "file3.txt", "irrelevant")
+	next(configuration)
+	start(configuration)
+
+	commits := commitsOnCurrentBranch(configuration)
+
+	equals(t, []string{
+		configuration.WipCommitMessage,
+		"on branch",
+	}, commits)
+}
+
+func TestEndsWithWipCommit_finalManualCommit(t *testing.T) {
+	_, configuration := localSetup(t)
+	start(configuration)
+	createFileAndCommitIt(t, "file1.txt", "irrelevant", "new file")
+
+	equals(t, false, endsWithWipCommit(configuration))
+}
+
+func TestEndsWithWipCommit_finalWipCommit(t *testing.T) {
+	_, configuration := localSetup(t)
+	start(configuration)
+	createFile(t, "file1.txt", "irrelevant")
+	next(configuration)
+	start(configuration)
+
+	equals(t, true, endsWithWipCommit(configuration))
+}
+
+func TestEndsWithWipCommit_manualThenWipCommit(t *testing.T) {
+	_, configuration := localSetup(t)
+	start(configuration)
+	createFileAndCommitIt(t, "file1.txt", "irrelevant", "new file")
+	createFile(t, "file2.txt", "irrelevant")
+	next(configuration)
+	start(configuration)
+
+	equals(t, true, endsWithWipCommit(configuration))
+}
+
+func TestEndsWithWipCommit_wipThenManualCommit(t *testing.T) {
+	_, configuration := localSetup(t)
+	start(configuration)
+	createFile(t, "file2.txt", "irrelevant")
+	next(configuration)
+	start(configuration)
+	createFileAndCommitIt(t, "file1.txt", "irrelevant", "new file")
+
+	equals(t, false, endsWithWipCommit(configuration))
+}
+
+func TestMarkSquashWip_singleManualCommit(t *testing.T) {
+	configuration = getDefaultConfiguration()
+	input := "pick c51a56d new file\n" +
+		"\n" +
+		"# Rebase ..."
+
+	result := markPostWipCommitsForSquashing(input, configuration)
+
+	equals(t, input, result)
+}
+
+func TestMarkSquashWip_manyManualCommits(t *testing.T) {
+	configuration = getDefaultConfiguration()
+	input := "pick c51a56d new file\n" +
+		"pick 63ef7a4 another commit\n" +
+		"\n" +
+		"# Rebase ..."
+
+	result := markPostWipCommitsForSquashing(input, configuration)
+
+	equals(t, input, result)
+}
+
+func TestMarkSquashWip_wipCommitFollowedByManualCommit(t *testing.T) {
+	configuration = getDefaultConfiguration()
+	input := "pick 01a9a31 " + configuration.WipCommitMessage + "\n" +
+		"pick c51a56d manual commit\n" +
+		"\n" +
+		"# Rebase ..."
+	expected := "pick 01a9a31 " + configuration.WipCommitMessage + "\n" +
+		"squash c51a56d manual commit\n" +
+		"\n" +
+		"# Rebase ..."
```

Comment:
please use multi-line strings for these cases. see https://yourbasic.org/golang/multiline-string/ for an example

Reply:
did refactor to multiline strings